tests/v2/validation/clustertemplates/provisioning/provisioning_clustertemplate_test.go

```diff
@@ -0,0 +1,212 @@
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type ClusterTemplateProvisioningTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	session            *session.Session
+	provisioningConfig *provisioninginput.Config
+}
+
+func (ct *ClusterTemplateProvisioningTestSuite) TearDownSuite() {
+	ct.session.Cleanup()
+}
+
+func (ct *ClusterTemplateProvisioningTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	ct.session = testSession
+
+	ct.provisioningConfig = new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, ct.provisioningConfig)
+
+	client, err := rancher.NewClient("", testSession) //ct.client
+	require.NoError(ct.T(), err)
+
+	ct.client = client
+
+	if ct.provisioningConfig.RKE1KubernetesVersions == nil {
+		rke1Versions, err := kubernetesversions.ListRKE1AllVersions(ct.client)
+		require.NoError(ct.T(), err)
+
+		ct.provisioningConfig.RKE1KubernetesVersions = []string{rke1Versions[len(rke1Versions)-1]}
+	}
+
+	if ct.provisioningConfig.CNIs == nil {
+		ct.provisioningConfig.CNIs = []string{clustertemplates.CniCalico}
+	}
+
+}
+
+func (ct *ClusterTemplateProvisioningTestSuite) TestProvisioningRKE1ClusterWithClusterTemplate() {
+	log.Info("Create an rke template and creating a downstream node driver with the rke template.")
+
+	clusterTemplate, err := clustertemplates.CreateRkeTemplate(ct.client, nil)
+	require.NoError(ct.T(), err)
+
+	templateRevisonConfig := mgmt.ClusterTemplateRevision{
+		ClusterConfig: &mgmt.ClusterSpecBase{RancherKubernetesEngineConfig: &mgmt.RancherKubernetesEngineConfig{
+			Network: &mgmt.NetworkConfig{Plugin: ct.provisioningConfig.CNIs[0]},
+			Version: ct.provisioningConfig.RKE1KubernetesVersions[0],
+		},
+		},
+	}
+
+	clusterTemplateRevision, err := clustertemplates.CreateRkeTemplateRevision(ct.client, templateRevisonConfig, clusterTemplate.ID)
+	require.NoError(ct.T(), err)
+
+	rke1Cluster := clustertemplates.UpdateClusterTemplateClusterConfig(clusterTemplate.ID, clusterTemplateRevision.ID, nil)
+
+	clusterObj, clusterConfig, err := clustertemplates.CreateRke1ClusterWithTemplate(ct.client, rke1Cluster, ct.provisioningConfig)
+	require.NoError(ct.T(), err)
+
+	log.Info("Verifying the rke1 cluster comes up active.")
+	clusterConfig.KubernetesVersion = ct.provisioningConfig.RKE1KubernetesVersions[0]
+	provisioning.VerifyRKE1Cluster(ct.T(), ct.client, clusterConfig, clusterObj)
+}
+
+func (ct *ClusterTemplateProvisioningTestSuite) TestClusterTemplateEnforcementAsAdmin() {
+	log.Info("Enforcing cluster template while provisioning rke1 clusters")
+
+	clusterEnforcement, err := ct.client.Steve.SteveType(settings.ManagementSetting).ByID(clustertemplates.ClusterEnforcementSetting)
+	require.NoError(ct.T(), err)
+	_, err = settings.UpdateGlobalSettings(ct.client.Steve, clusterEnforcement, clustertemplates.EnabledClusterEnforcementSetting)
+	require.NoError(ct.T(), err)
```

Comment:
I just realized this is a global setting. will this affect other tests / does this persist after the test is finished, or does cleanup take care of setting it back to the default? Or am I misreading what this does?

Reply:
cleanup doesnt take care of this as its an existing setting that we are updating. but I added a logic to change the setting back to the original. As of now it doesnt effect any other tests as its the last one but the setting do persist if its not updated to the original value.